compiler/src/main/java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -950,7 +950,8 @@ void resolve(BindingKey bindingKey) {
          * 2. If there are any explicit bindings in this component, they may conflict with those in
          *    the supercomponent, so resolve them here so that conflicts can be caught.
          */
-        if (getPreviouslyResolvedBindings(bindingKey).isPresent()) {
+        if (getPreviouslyResolvedBindings(bindingKey).isPresent()
+            && bindingKey.kind().equals(BindingKey.Kind.CONTRIBUTION)) {
```

Comment:
I looked more into this and here's what I've got: 

```java
if (getPreviouslyResolvedBindings(bindingKey).isPresent()) {
  /* Resolve in the parent in case there are multibinding contributions or conflicts in some
   * component between this one and the previously-resolved one. */
  parentResolver.get().resolve(bindingKey);
  boolean localExplicit;
  if (bindingKey.kind().equals(BindingKey.Kind.CONTRIBUTION)) {
    localExplicit = getLocalExplicitBindings(bindingKey.key()).isEmpty();
  } else {
    localExplicit =
        getPreviouslyResolvedBindings(bindingKey)
            .get()
            .bindings()
            .stream()
            .flatMap(binding -> binding.dependencies().stream())
            .anyMatch(dependency -> getLocalExplicitBindings(dependency.key()).isEmpty());
  }
  if (!new LocalDependencyChecker().dependsOnLocalBindings(bindingKey) && localExplicit) {
```

I think that's working correctly, even if there are dependencies of the members injection binding that transitively have local multibinding contributions. @netdpb does something like this make sense to you?

We can obviously clean this up before submitting.